chain/store/name.go

```diff
@@ -123,24 +193,24 @@ func (cs *ChainStore) GetRegistrant(name string) ([]byte, error) {
 
 func (sdb *StateDB) FinalizeNames(commit bool) {
 	sdb.names.Range(func(key, value interface{}) bool {
-		if registrantId, ok := key.(string); ok {
-			if name, ok := value.(string); ok && len(name) > 0 {
-				nameId := getNameId(name)
-				if v, ok := sdb.nameRegistrants.Load(nameId); ok {
-					if registrant, ok := v.([]byte); ok {
-						sdb.updateName(registrant, name)
-						if commit {
-							sdb.nameRegistrants.Delete(nameId)
-						}
-					}
+		if nameId, ok := key.([]byte); ok {
+			if info, ok := value.(nameInfo); ok && !info.Empty() {
```

Comment:
It should be `*nameInfo` instead of `nameInfo`

Reply:
done